src/core/lib/security/credentials/fake/fake_credentials.c

```diff
@@ -49,7 +50,22 @@ static grpc_security_status fake_transport_security_create_security_connector(
     grpc_call_credentials *call_creds, const char *target,
     const grpc_channel_args *args, grpc_channel_security_connector **sc,
     grpc_channel_args **new_args) {
-  *sc = grpc_fake_channel_security_connector_create(call_creds);
+  if (grpc_channel_args_find(args, GRPC_ARG_GRPCLB_BALANCER_NAMES) != NULL) {
+    /* For LB channels, verify the target name passed in matches the
```

Comment:
If the goal here is that the security connector code will lookup the target name from GRPC_ARG_GRPCLB_BALANCER_NAMES, then why do we need to check the target name passed to the security connector?  It seems like we shouldn't actually care about that.